tensorboard/components/tf_ng_tensorboard/reloader/reloader.component.ts

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy} from '@angular/core';
+import {Store, select} from '@ngrx/store';
+import {combineLatest} from 'rxjs';
+import {distinctUntilChanged} from 'rxjs/operators';
+
+import {
+  State,
+  getReloadEnabled,
+  getReloadPeriodInMs,
+} from '../core/core.reducers';
+import {reload} from '../core/core.actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'reloader',
```

Comment:
(i might be totally wrong) but if there's no plan to use the actual DOM Element, should this be a ReloaderService provided to the main AppComponent?

also, can we drop the template: ''?

Reply:
I really hope to make our codebase more like Redux instead of buying into Angular's service-based architecture. In other words, I do not want to see the store injected into any service if possible. In React world, DOM-less functional component is a pattern that I seldom employed. It was basically for enabling/disabling a feature in declarative manner.

Dunno why Angular insisted that we have it but it complains without the template.

Reply:
Thanks for explaining this!  While it isn't interfering with the app, I guess these functional components in DOM are okay, as long as they stay at the root level.